src/cpp/database/database.hpp

```diff
@@ -306,6 +306,15 @@ class Database
     void load_database(
             const DatabaseDump& dump);
 
+    /**
+     * Change the status (active/inactive) of an entity given an EntityId.
+     * Also check if the references of the entity must also be changed and change the status in that case.
+     * @throws eprosima::statistics_backend::BadParameter if entity_kind is not valid.
```

Comment:
We came to the conclusion of change the method to `noexcept`.
I don't remove the @throws doc of the public method because the call to get_entity_kind can throw an exception.
I remove it from the auxiliar method because now is `noexcept`

Reply:
After some more discussion we decided to revert this to its original state: Leave both methods with the @throw annotation, as it happens that both can finally throw.

Reply:
I leave it as it is, we decided that the aux does not throw any exception

Reply:
I leave it as it is, we decided that the aux does not throw any exception